plugin/src/main/java/org/opensearch/ml/sdkclient/DDBOpenSearchClient.java

```diff
@@ -321,14 +352,36 @@ private XContentParser createParser(String json) throws IOException {
         return jsonXContent.createParser(NamedXContentRegistry.EMPTY, DeprecationHandler.IGNORE_DEPRECATIONS, json);
     }
 
-    private String simulateOpenSearchResponse(String index, String id, String source, Map<String, Object> additionalFields) {
+    private GetItemRequest getGetItemRequest(String requestTenantId, String documentId, String index) {
+        final String tenantId = requestTenantId != null ? requestTenantId : DEFAULT_TENANT;
+        final GetItemRequest getItemRequest = GetItemRequest
+            .builder()
+            .tableName(getTableName(index))
+            .key(
+                Map
+                    .ofEntries(
+                        Map.entry(HASH_KEY, AttributeValue.builder().s(tenantId).build()),
+                        Map.entry(RANGE_KEY, AttributeValue.builder().s(documentId).build())
+                    )
+            )
+            .build();
+        return getItemRequest;
+    }
+
+    private String simulateOpenSearchResponse(
+        String index,
+        String id,
+        String source,
+        Long sequenceNumber,
+        Map<String, Object> additionalFields
+    ) {
         StringBuilder sb = new StringBuilder("{");
         // Fields with a DDB counterpart
         sb.append("\"_index\":\"").append(index).append("\",");
         sb.append("\"_id\":\"").append(id).append("\",");
         // Fields we must simulate using default values
-        sb.append("\"_primary_term\":").append(UNASSIGNED_PRIMARY_TERM).append(",");
-        sb.append("\"_seq_no\":").append(UNASSIGNED_SEQ_NO).append(",");
+        sb.append("\"_primary_term\":").append(sequenceNumber == null ? UNASSIGNED_PRIMARY_TERM : 1).append(",");
+        sb.append("\"_seq_no\":").append(sequenceNumber == null ? UNASSIGNED_SEQ_NO : sequenceNumber).append(",");
```

Comment:
(Readability) Might be cleaner to wrap these two lines in an `if (sequenceNumber == null)` conditional to make the pairing more clear (i.e., unassigned for both, or seqNo >=0 + primaryTerm=1)